pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelay = 10
+     )
+ 
+     $attempt = 0
+ 
+     while ($attempt -lt $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying..."
```

Comment:
>                  Write-Host "Attempt $attempt: error getting HNS Network. Retrying in $retryDelay seconds..."

Consider rewording the message to capture the failed operation and providing feedback on when the next attempt will takes place

Reply:
the failure information is captured on the catch. Nothing is really failing here, so there's nothing to capture.

Reply:
by "failed operation " I meant the `null` response.